storage/remote/queue_manager.go

```diff
@@ -1921,11 +1925,18 @@ func populateV2TimeSeries(symbolTable *writev2.SymbolsTable, batch []timeSeries,
 			pendingData[nPending].Metadata.UnitRef = symbolTable.Symbolize(d.metadata.Unit)
 			nPendingMetadata++
 		} else {
+			metricType := writev2.Metadata_METRIC_TYPE_UNSPECIFIED
+			var unit string
+			if enableTypeAndUnitLabels {
+				metadata := schema.NewMetadataFromLabels(d.seriesLabels)
+				metricType = modelTypeToWriteV2Type(string(metadata.Type))
+				unit = metadata.Unit
+			}
 			// Safeguard against sending garbage in case of not having metadata
 			// for whatever reason.
-			pendingData[nPending].Metadata.Type = writev2.Metadata_METRIC_TYPE_UNSPECIFIED
+			pendingData[nPending].Metadata.Type = metricType
 			pendingData[nPending].Metadata.HelpRef = 0
-			pendingData[nPending].Metadata.UnitRef = 0
+			pendingData[nPending].Metadata.UnitRef = symbolTable.Symbolize(unit)
```

Comment:
Nice! This could be probably simplified to:

```
var m schema.Metadata
if enableTypeAndUnitLabels {
			m = schema.NewMetadataFromLabels(d.seriesLabels)
}
pendingData[nPending].Metadata.Type = writev2.FromMetadataType(m.Type)
pendingData[nPending].Metadata.UnitRef = symbolTable.Symbolize(m.Unit)
pendingData[nPending].Metadata.HelpRef = 0
```

WDYT?

Reply:
[172f028](https://github.com/prometheus/prometheus/pull/17033/commits/172f0280ec7e8991f49537860479d5262102467c)